pandas/core/ops.py

```diff
@@ -264,7 +263,57 @@ def add_flex_arithmetic_methods(cls, flex_arith_method, radd_func=None,
                 exclude=exclude)
 
 
-class _TimeOp(object):
+class _Op(object):
+
+    fill_value = np.nan
+    wrap_results = staticmethod(lambda x: x)
+    dtype = None
+
+    def __init__(self, left, right, name, na_op):
+
+        self.left = left
+        self.right = right
+
+        self.name = name
+        self.na_op = na_op
+
+        self.lvalues = left
+        self.rvalues = right
+
+    @classmethod
+    def get_op(cls, left, right, name, na_op):
+        """
+        Get op dispatcher, returns _ArithOp or _TimeOp.
+
+        if ``left`` and ``right`` are appropriate for datetime arithmetic with
```

Comment:
ahh I see you did it here, ok maybe just move this before ``__init__`` as then you read it first and get whats going on.